include/exadg/convection_diffusion/spatial_discretization/operator.cpp

```diff
@@ -88,14 +90,29 @@ Operator<dim, Number>::initialize_dof_handler_and_constraints()
   }
 
   affine_constraints.close();
+}
 
-  pcout << std::endl
-        << "Discontinuous Galerkin finite element discretization:" << std::endl
-        << std::endl;
+template<int dim, typename Number>
+void
+Operator<dim, Number>::print_discretization_info(bool const verbosity) const
+{
+  pcout << std::endl << "Discontinuous Galerkin finite element discretization";
+  if(param.enable_adaptivity)
+  {
+    pcout << " (adaptivity enabled):";
+  }
+  pcout << std::endl << std::endl;
 
-  print_parameter(pcout, "degree of 1D polynomials", param.degree);
-  print_parameter(pcout, "number of dofs per cell", fe->n_dofs_per_cell());
-  print_parameter(pcout, "number of dofs (total)", dof_handler.n_dofs());
+  if(verbosity)
+  {
+    print_parameter(pcout, "degree of 1D polynomials", param.degree);
+    print_parameter(pcout, "number of dofs per cell", fe->n_dofs_per_cell());
+    print_parameter(pcout, "number of dofs (total)", dof_handler.n_dofs());
+  }
+  else
+  {
+    print_parameter(pcout, "number of dofs (total)", dof_handler.n_dofs());
```

Comment:
can't we just write this line in `setup_after_coarsening_and_refinement()` and then skip all the parameters (enable_adaptivity, verbosity) here? I think this would be more simple overall.

Reply:
restructured it!